lib/wallet/api/http/Cardano/Wallet/Api/Types/RestorationMode.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE LambdaCase #-}
+
+module Cardano.Wallet.Api.Types.RestorationMode
+    ( RestorationMode (..)
+    ) where
+
+import Prelude
+
+import Data.Aeson
+    ( FromJSON (parseJSON)
+    , KeyValue ((.=))
+    , ToJSON (toJSON)
+    , Value (..)
+    , object
+    , (.:)
+    )
+import Data.Aeson.Types
+    ( Parser
+    )
+import Data.Quantity
+    ( Quantity (Quantity)
+    )
+import Data.Time
+    ( UTCTime
+    )
+import Data.Time.Format.ISO8601
+    ( iso8601ParseM
+    , iso8601Show
+    )
+import Data.Word
+    ( Word32
+    )
+import GHC.Generics
+    ( Generic
+    )
+
+import qualified Data.Text as T
+
+data RestorationMode
+    = RestoreFromGenesis
+    | RestoreFromSlot (Quantity "slot" Word32)
+    | RestoreFromTip
+    | RestoreFromDate UTCTime
+    | RestoreFromBlock (Quantity "block" Word32)
+    deriving (Eq, Show, Generic)
+
+instance ToJSON RestorationMode where
+    toJSON :: RestorationMode -> Value
+    toJSON = \case
+        RestoreFromGenesis -> object [ "unit" .= String "genesis" ]
```

Comment:
if we change as I proposed in swagger then of course here we would just have `String "from_genesis"` .... etc